### PR TITLE
Add appendRowNumber option to presto-hive

### PR DESCRIPTION
**Pull request by harsharastogi**
OrcSelectiveReader now supports appending rowNumber column to a page. This change allows
enabling that option through the hive selective page source factory 

Test plan
Ensure all existing tests pass

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveSelectivePageSourceFactory.java [hunk 1]
```diff
@@ -23,6 +23,8 @@
 import org.apache.hadoop.fs.Path;
 import org.joda.time.DateTimeZone;
 
+import javax.swing.text.html.Option;
```
Comment (arunthirupathi) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveSelectivePageSourceFactory.java:26]: wrong import.

File: presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetSelectivePageSourceFactory.java [hunk 2]
```diff
@@ -71,7 +71,8 @@ public ParquetSelectivePageSourceFactory()
             RowExpression remainingPredicate,
             DateTimeZone hiveStorageTimeZone,
             HiveFileContext hiveFileContext,
-            Optional<EncryptionInformation> encryptionInformation)
+            Optional<EncryptionInformation> encryptionInformation,
+            boolean appendRowNumberEnabled)
```
Comment (arunthirupathi) [inline presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetSelectivePageSourceFactory.java:75]: add a checkState, that appendRowNumberEnabled is not supported here.

File: presto-hive/src/main/java/com/facebook/presto/hive/HivePartialAggregationPushdown.java [hunk 3]
```diff
@@ -263,7 +263,8 @@ private boolean isAggregationPushdownSupported(AggregationNode partialAggregatio
                     oldTableLayoutHandle.isPushdownFilterEnabled(),
                     oldTableLayoutHandle.getLayoutString(),
                     oldTableLayoutHandle.getRequestedColumns(),
-                    true);
+                    true,
+                    false);
```
Comment (arunthirupathi) [inline presto-hive/src/main/java/com/facebook/presto/hive/HivePartialAggregationPushdown.java:267]: why are we passing in false for the appendRowNumber here. If this should be false, should we checkState oldTableLayoutHandle.getAppendRowNumber() is false.

**Review by arunthirupathi — COMMENTED** [review]
Mostly looks good.

File: presto-hive/src/main/java/com/facebook/presto/hive/HivePartialAggregationPushdown.java (hunk 3 above)
Comment (harsharastogi) [reply presto-hive/src/main/java/com/facebook/presto/hive/HivePartialAggregationPushdown.java:267]: Good catch, it should pass oldTableLayoutHandle.isAppendRowNumber()

File: presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdown.java [hunk 4]
```diff
@@ -294,6 +294,7 @@ public static ConnectorPushdownFilterResult pushdownFilter(
                                         remainingExpression,
                                         domainPredicate),
                                 currentLayoutHandle.map(layout -> ((HiveTableLayoutHandle) layout).getRequestedColumns()).orElse(Optional.empty()),
+                                false,
                                 false)),
```
Comment (arunthirupathi) [inline presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdown.java:298]: Should this be extracted from layout as well ?

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java [hunk 5]
```diff
@@ -2706,6 +2706,7 @@ public List<ConnectorTableLayoutResult> getTableLayouts(ConnectorSession session
                                 false,
                                 createTableLayoutString(session, handle.getSchemaTableName(), hivePartitionResult.getBucketHandle(), hivePartitionResult.getBucketFilter(), TRUE_CONSTANT, domainPredicate),
                                 desiredColumns.map(columns -> columns.stream().map(column -> (HiveColumnHandle) column).collect(toImmutableSet())),
+                                false,
                                 false)),
```
Comment (arunthirupathi) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java:2710]: Can this always be false ?

**Review by arunthirupathi — APPROVED** [review]
Can you please format the commit message (not PR) to be compliant with Presto standards.

72 chars per line on the commit message and a smaller heading.

Once the comments and the commit message is altered this PR is good to be merged.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java (hunk 5 above)
Comment (harsharastogi) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java:2710]: I think its safer to put it false for now

File: presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdown.java (hunk 4 above)
Comment (arunthirupathi) [reply presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdown.java:298]: I am confused, appendRowNumber is always set to false.  but looking at the code above, requestedColumns is extracted from currentLayoutHandle, should the appendRowNumber be extracted from currentLayoutHandle as well ?

**Comment by harsharastogi** [comment]
> Can you please format the commit message (not PR) to be compliant with Presto standards.
> 
> 72 chars per line on the commit message and a smaller heading.
> 
> Once the comments and the commit message is altered this PR is good to be merged.

Done

File: presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdown.java (hunk 4 above)
Comment (harsharastogi) [reply presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdown.java]: Yes, updated it to inherit from currentLayoutHandle

**Review by arunthirupathi — APPROVED** [review]

**Pull request merged**